ring/partition_ring_model.go

```diff
@@ -133,6 +133,7 @@ func (m *PartitionRingDesc) AddPartition(id int32, state PartitionState, now tim
 	spreadMinimizing := NewSpreadMinimizingTokenGeneratorForInstanceAndZoneID("", int(id), 0, false)
 
 	m.Partitions[id] = PartitionDesc{
+		Id:             id,
```

Comment:
This will only work for newly added partitions. What about partitions that already exist in the ring (like on our dev cluster), how will we make sure that those get correct `Id` as well?

Reply:
For instance-ring, we covered that by https://github.com/grafana/dskit/blob/5d716e13ec96347b7196e89087b954705a0369b3/ring/ring.go#L313-L316 for readers of the ring. We could do something similar in `PartitionRingWatcher`.

Reply:
> This will only work for newly added partitions. What about partitions that already exist in the ring (like on our dev cluster), how will we make sure that those get correct Id as well?

Do we really need to support it? Given it's under development, my idea was to simply ignore the migration problem.

Reply:
I checked the actual implementation and adding support for migration is not difficult, but looks too weird to me that we have to migrate for a feature which is still in development. I will just re-create it in the dev cell.